homeassistant/components/light/rpi_gpio_pwm.py

```diff
@@ -95,15 +97,26 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     add_entities(leds)
 
 
-class PwmSimpleLed(Light):
+class PwmSimpleLed(Light, RestoreEntity):
     """Representation of a simple one-color PWM LED."""
 
     def __init__(self, led, name):
         """Initialize one-color PWM LED."""
         self._led = led
         self._name = name
         self._is_on = False
-        self._brightness = 255
+        self._brightness = DEFAULT_BRIGHTNESS
+
+    async def async_added_to_hass(self):
+        """Handle entity about to be added to hass event."""
+        await super().async_added_to_hass()
+        last_state = await self.async_get_last_state()
+        if last_state:
+            self._is_on = (last_state.state == STATE_ON)
+            self._brightness = last_state.attributes.get('brightness',
+                                                         DEFAULT_BRIGHTNESS)
+            self._led.set(is_on=self._is_on,
```

Comment:
Is this async safe, ie no I/O or blocking?

Reply:
No, this one does I/O. How should this be handled? Should it be executed via `hass.async_add_executor_job`?

Reply:
Yes!

Reply:
Please open a new PR with those changes.

Reply:
Alright, fixed this here: https://github.com/home-assistant/home-assistant/pull/20172